ratschlab_common/io/sparse_df.py

```diff
@@ -0,0 +1,122 @@
+
+def _read_sparse_m(group):
+    attributes = []
+    for attribute in ('data', 'indices', 'indptr', 'shape'):
+        attributes.append(getattr(group, attribute).read())
+    sparse_m = sparse.csr_matrix(tuple(attributes[:3]), shape=attributes[3])
+    return sparse_m
+
+def _read_sparse_m_coo(group):
+    attributes = []
+    for attribute in ('data', 'row', 'col', 'shape'):
+        attributes.append(getattr(group, attribute).read())
+    t = tuple([attributes[1], attributes[2]])
+    sparse_m = sparse.coo_matrix(tuple([attributes[0], t]), shape=attributes[3])
+    return sparse_m
+
+def _store_sparse_m_coo(m, hdf5, group, complevel=5, complib='blosc', **kwargs):
+    assert (m.__class__ == sparse.coo.coo_matrix), 'm must be a coo matrix'
+    for attribute in ('data', 'row', 'col', 'shape'):
+        arr = np.array(getattr(m, attribute))
+        _store_array_hdf(arr, hdf5, group, attribute,
+                                        complevel=complevel, complib=complib, **kwargs)
+    return hdf5
+
+
+def _store_sparse_m(m, hdf5, group, complevel=5, complib='blosc', **kwargs):
+    assert (m.__class__ == sparse.csr.csr_matrix), 'M must be a csr matrix'
+    for attribute in ('data', 'indices', 'indptr', 'shape'):
+        arr = np.array(getattr(m, attribute))
+        _store_array_hdf(arr, hdf5, group, attribute,
+                                        complevel=complevel, complib=complib, **kwargs)
+    return hdf5
+
+def _store_array_hdf(arr, hdf5, group, name, complevel=5, complib='blosc', **kwargs):
+    if arr.dtype == 'object':
+        arr = arr.astype('U')
+    atom = tables.Atom.from_dtype(arr.dtype)
+    filters = tables.Filters(complevel=complevel, complib=complib, **kwargs)
+    ds = hdf5.create_carray(group, name, atom, arr.shape, filters=filters)
+    ds[:] = arr
+
+def _store_non_sparse_df(df, path, complevel=5, complib='blosc', **kwargs):
+    df.to_hdf(path, key="data/non_sparse",
+                         complevel=complevel, complib=complib, **kwargs)
+
+
+def to_hdf(df, path, **kwargs):
+    '''
+    write sparse dataframe to hdf file . The default compression level and complib are:
+    complevel: 5
+    complib: 'blosc'
+    If you wish to change that, pass a dictionary to the function call as in the following example:
+    pars = {
+     'sparse': {
+      'complevel': 6,
+      'complib': 'zlib'
+     }
+     'non_sparse': {
+      'complevel': 7,
+      'complib': 'lzo'
+     }
+    }
+    to_hdf(my_df, my_path, **pars).
+
+    The sparse matrix is saved in COO format
+    :param df: an instance of Pandas DataFrame that you wish to save to file
+    :param path: path of the
+    :param kwargs:
+    :return:
+    '''
+    kwargs_sparse = kwargs.get('sparse', {})
+    kwargs_non_sparse = kwargs.get('non_sparse', {})
+    assert isinstance(kwargs_sparse, dict) and isinstance(kwargs_non_sparse, dict)
+
+    sparse_df, non_sparse_df = _split_sparse(df)
+    sparse_m = sparse.coo_matrix(sparse_df.values)
```

Comment:
I think this should be `sparse_df.sparse.to_coo()`. `sparse_df.values` will convert it to a dense array which sort of defeats the whole purpose...